### DISCUSSION TITLE
How do I make queries accessible in workflow context?

### DISCUSSION THREAD

**Discussion by weirdyang** [OrchardCMS/OrchardCore#15436] [Q&A] [answered]
I would like to make a module and have the queries accessible in workflows, sort of how we can use SiteSettings.NameOfSetting, is this possible?

**Comment by douwinga** [comment]
You shouldn't need a custom module for this as workflows already can support this. You can use the "Set Property Task" with the javascript function `executeQuery` for instance.
https://docs.orchardcore.net/en/latest/docs/reference/modules/Workflows/#javascript-functions
![image](https://github.com/OrchardCMS/OrchardCore/assets/11961383/2377e8cb-929c-4b50-92f5-7a660a13ec24)

You can also use liquid to do the query
https://docs.orchardcore.net/en/latest/docs/reference/modules/Queries/#query_1
![image](https://github.com/OrchardCMS/OrchardCore/assets/11961383/6cc62699-1a67-403a-9327-54edfd295cea)

**Reply by weirdyang** [reply]
thanks, apologies as my question wasn't clear, I was wondering how did the query module implement these features. I would like to implement a custom queries module so I can query other databases besides the orchardcore one.

**Comment by weirdyang** [comment, marked as answer]
Ok so I found my answer by going into the code and the docs:
![image](https://github.com/OrchardCMS/OrchardCore/assets/25576022/028559c7-4b9c-4114-849d-056187eb6404)
Configure template options to add the necessary services and define the name to access it by, the docs was useful in explaining this piece of code

```

            services.Configure<TemplateOptions>(o =>
            {
                o.Scope.SetValue("Queries", new ObjectValue(new LiquidQueriesAccessor()));
                o.MemberAccessStrategy.Register<LiquidQueriesAccessor, FluidValue>(async (obj, name, context) =>
                {
                    var liquidTemplateContext = (LiquidTemplateContext)context;
                    var queryManager = liquidTemplateContext.Services.GetRequiredService<IQueryManager>();

                    return FluidValue.Create(await queryManager.GetQueryAsync(name), context.Options);
                });
            })
            .AddLiquidFilter<QueryFilter>("query");
  ```
